File: pipeline/archivebot/shared_config.py
import os
import yaml
try:
	from yaml import CLoader as Loader
except ImportError:
	from yaml import Loader

def config():
    my_dir = os.path.dirname(__file__)
    config_file = os.path.join(my_dir, '../../lib/shared_config.yml')

    with open(config_file, 'r') as f:
        return yaml.load(f.read(), Loader = Loader)

def log_channel():
    c = config()

    return c['channels']['log']

def pipeline_channel():
    c = config()

    return c['channels']['pipeline']

def job_channel(ident):
    return '%s%s' % (job_channel_prefix(), ident)

def job_channel_prefix():
    c = config()

    return c['channels']['job_prefix']

# vim:ts=4:sw=4:et:tw=78

File: lib/shared_config.rb
require 'yaml'

module SharedConfig
  module_function

  def config
    # For maximum fun, check out https://github.com/tenderlove/psych/issues/119
    # re: the development of safe_load
    YAML.safe_load(File.read(File.expand_path('../shared_config.yml', __FILE__)))
  end

  def log_channel
    config['channels']['log']
  end

  def job_channel(ident)
    "#{job_channel_prefix}#{ident}"
  end

  def job_channel_prefix
    config['channels']['job_prefix']
  end

  def irc_server_uri
    config['irc']['server_uri']
  end

  def irc_nick
    config['irc']['nick']
  end

  def irc_channels
    puts config['irc'].inspect
    config['irc']['channels']
  end
end
